spec/policy/scope_check/scope_check_spec.lua

```diff
@@ -0,0 +1,534 @@
+        end)
+
+        it('client role', function()
+          local scope_check_policy = ScopeCheckPolicy.new({
+            scopes = {
+              {
+                client_roles = { { name = "aaa", client = "ccc" } },
+                resource = "/bbb"
+              }
+            }
+          })
+
+          ngx.var = {
+            uri = '/bbb'
+          }
+
+          local context = {
+            jwt = {
+              resource_access = {
+                ccc = {
+                  roles = { "aaa" }
+                }
+              }
+            }
+          }
+
+          assert(scope_check_policy:access(context))
+        end)
+
+        it('multi roles in policy', function()
+          local scope_check_policy = ScopeCheckPolicy.new({
+            scopes = {
+              {
+                realm_roles = { { name = "ddd" } },
+                client_roles = { { name = "aaa", client = "ccc" } },
+                resource = "/bbb"
+              }
+            }
+          })
+
+          ngx.var = {
+            uri = '/bbb'
+          }
+
+          local context = {
+            jwt = {
+              realm_access = {
+                roles = {"ddd", "eee"}
+              },
+              resource_access = {
+                ccc = {
+                  roles = { "aaa" }
+                }
+              }
+            }
+          }
+
+          assert(scope_check_policy:access(context))
+        end)
+
+        it('wildcard', function()
+          local scope_check_policy = ScopeCheckPolicy.new({
+            scopes = {
+              {
+                client_roles = { { name = "aaa", client = "ccc" } },
+                resource = "/bbb/{wildcard}"
+              }
+            }
+          })
+
+          ngx.var = {
+            uri = '/bbb/ddd'
+          }
+
+          local context = {
+            jwt = {
+              resource_access = {
+                ccc = {
+                  roles = { "aaa", "eee" }
+                }
+              }
+            }
+          }
+
+          assert(scope_check_policy:access(context))
+        end)
+
+        it('multi scopes', function()
+          local scope_check_policy = ScopeCheckPolicy.new({
+            scopes = {
+              {
+                client_roles = { { name = "aaa", client = "ccc" } },
+                resource = "/bbb/{wildcard}"
+              },
+              {
+                realm_roles = { { name = "rrr" } },
+                client_roles = { { name = "fff", client = "hhh" } },
+                resource = "/eee{wildcard}"
+              },
+              {
+                realm_roles = { { name = "ddd" } },
+                resource = "/eeeyyy{wildcard}"
+              },
+              {
+                realm_roles = { { name = "rrr" } },
+                resource = "/{wildcard}ggg"
+              },
+              {
+                client_roles = { { name = "fff", client = "hhh" } },
+                resource = "/z{wildcard}ggg"
```

Comment:
Those tests are not very readable to me. I'm not sure what is it, but it is hard to understand what is going on.

Maybe it is just those random characters in roles and urls, or actually that they are quite big and have a lot of configuration. It is hard to know *why* the test should be passing/failing.

Reply:
Changed.